src/plugins/tangorin.rs

```diff
@@ -0,0 +1,214 @@
+                else{
+                    if let Some(text_) = self.inner_text(&child_) {
+                        meaning.push_str(text_.as_str());
+                    }
+                }
+            }
+
+           Some(meaning)
+        }
+        else{
+            None
+        }
+    }
+
+    fn retrieve_info(&self, doc: &kuchiki::NodeRef) -> Option<String> {
+        if let Some(match_) = doc.select("span[class=eng]").unwrap().next() {
+            let node = match_.as_node();
+            let following_siblings = node.following_siblings();
+
+            if let Ok(mut info_sibs) = following_siblings.select("i[class=d-info]") {
+                match info_sibs.next() {
+                    Some(first_sibling) => self.inner_text(first_sibling.as_node()),
+                    None => None
+                }
+            }
+            else{
+                None
+            }
+        }
+        else {
+            None
+        }
+    }
+
+    fn inner_text(&self, root: &kuchiki::NodeRef) -> Option<String> {
+        if let Some(match_) = root.first_child() {
+            let res = match match_.as_text() {
+                Some(result_str) => result_str.borrow(),
+                None => { return None }
+            };
+            Some(res.clone())
+        }
+        else {
+            None
+        }
+    }
+
+    fn tangorin(&self, server: &IrcServer, _: &Message, target: &str, msg: &str) -> io::Result<()> {
+        let url = match self.grep_kanji(msg) {
+            Some(kanji) => "http://tangorin.com/general/".to_string() + kanji.as_str(),
+            None      => { return Ok(()); }
+        };
+
+        if let Ok(doc) = kuchiki::parse_html().from_http(&url) {
+            let romaji = match self.retrieve_from_selector(&doc, "rt"){
+                Some(retrieved) => retrieved,
+                None => { return Ok(()); } 
+            };
+            
+            let kana = match self.retrieve_from_selector(&doc, "rb") {
+                Some(retrieved) => retrieved,
+                None => { return Ok(()); } 
+            };
+         
+            let kanji = match self.retrieve_from_selector(&doc, "span[class=writing]") {
+                Some(retrieved) => retrieved,
+                None => { return Ok(()); }
+            };
+         
+            let meaning = match self.retrieve_meaning(&doc) {
+                Some(retrieved) => retrieved,
+                None => { return Ok(()); }
+            };
+
+            let info : String = match self.retrieve_info(&doc) {
+                Some(retrieved) => {
+                    let sanitized = retrieved.replace("\u{2014}", "").replace(".", "").to_lowercase();
+                    (" (".to_string()+sanitized.as_str()+")").to_string()
+                },
+                None => "".to_string()
+            };
+          
+            return server.send_privmsg(target, &format!("[Tangorin] {} ({} - {}): {}{}", &*kanji, &*kana, &*romaji, &*meaning, &*info))
```

Comment:
I'm pretty sure that filling a proper struct with these data and using the `ToString` trait would work great